extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/HttpCommonTags.java

```diff
@@ -46,29 +48,43 @@ public static Tag outcome(int statusCode) {
 
     /**
      * Creates a {@code uri} tag based on the URI of the given {@code request}.
-     * Falling back to {@code REDIRECTION} for 3xx responses, {@code NOT_FOUND}
-     * for 404 responses, {@code root} for requests with no path info, and {@code UNKNOWN}
+     * Falling back to {@code REDIRECTION} for 3xx responses if there wasn't a matched path pattern, {@code NOT_FOUND}
+     * for 404 responses if there wasn't a matched path pattern, {@code root} for requests with no path info, and
+     * {@code UNKNOWN}
      * for all other requests.
      *
      * @param pathInfo request path
+     * @param initialPath initial path before request pattern matching took place. Pass in null if there is pattern matching
+     *        done in the caller.
      * @param code status code of the response
      * @return the uri tag derived from the request
      */
-    public static Tag uri(String pathInfo, int code) {
-        if (code > 0) {
-            if (code / 100 == 3) {
-                return URI_REDIRECTION;
-            } else if (code == 404) {
-                return URI_NOT_FOUND;
-            }
-        }
+    public static Tag uri(String pathInfo, String initialPath, int code) {
         if (pathInfo == null) {
             return URI_UNKNOWN;
         }
         if (pathInfo.isEmpty() || "/".equals(pathInfo)) {
             return URI_ROOT;
         }
 
+        if (code > 0) {
+            if (code / 100 == 3) {
+                if (initialPath != null && !Objects.equals(initialPath, pathInfo)) {
```

Comment:
move this to a method that returns a boolean? You're doing it twice, and it needs a comment, and it involves some nots... 

That method could either be called to pass a boolean parameter, or in here.. but to prevent future accident between 3xx and 4xx.. I would recommend a tiny method.

Reply:
Thank you for the feedback. I've updated my commit and force-pushed the change.